app/helpers/sidebar_helper.rb

```diff
@@ -18,10 +18,15 @@ def menu_item(label:, path:, visible: false)
   private # private doesn't work in modules. It's here for semantic purposes
 
   def active_class(link_path)
-    url_route_sections = link_path.split("/")
-    url_route_sections.delete("all_casa_admins")
-    controller_name = url_route_sections.second
-    current_page?({controller: controller_name, action: action_name}) ? "active" : ""
+    if request.path == link_path
+      "active"
+    elsif link_path == "/emancipation_checklists" && request.path.match("emancipation")
+      # This check is needed because Sidebar menu item 'Emancipation Checklist(s)'
+      # contains a redirect if any @casa_transitioning_cases are found
+      "active"
+    else
+      ""
+    end
```

Comment:
The codeclimate error is complaining about too much complexity. How about this? (keeping your comment in there somewhere)

```
if (request.path == link_path) || (link_path == "/emancipation_checklists" && request.path.match("emancipation"))
  "active"
else
 ""
end
rescue ActionController::UrlGenerationError
  ""
end
```